src/test/compile-middleware_test.ts

```diff
@@ -109,4 +109,32 @@ suite('compile-middleware', () => {
       assert.notInclude(response.text, `<script>\nclass A {}\n</script>`);
     });
   });
+
+  test('needCompilation', () => {
+    const cases: {userAgent: string, expected: boolean}[] = [
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/14.99999',
+        expected: true,
+      },
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.14986',
+        expected: true,
+      },
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.15063',
+        expected: false,
+      },
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/16.00000',
+        expected: false,
+      },
+    ];
+    for (const c of cases) {
+      assert.equal(needCompilation(c.userAgent), c.expected, c.userAgent);
```

Comment:
Yeah, I agree this table-driven testing pattern could lead to bad failure messages, but I do personally find it kind of straightforward to read. Maybe I'm just really used to it because I've been writing so much Go, though, where it's really common.

Latest commit has a slightly more concise version now that we can use destructuring. Happy to change to the function style if you prefer it, though.